src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +379,168 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    public void validateSummaryTag(DetailNode ast) {
+        if (getInlineSummary(ast).isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            final String inlineSummaryDoc = getInlineSummary(ast);
+            if (isPeriodAtEnd(inlineSummaryDoc, period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
+            else if (isSummaryValid(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
+        }
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
+        final String summarySentence = sentence.trim();
+        return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
+    }
+
+    /**
+     * Get summary sentence for inline type summary javadoc.
+     *
+     * @param javadoc javadoc root node.
+     * @return  Summary sentence of javadoc root node.
+     */
+    private static String getInlineSummary(DetailNode javadoc) {
+        final String summary;
+        final DetailNode descriptionNode = getDescriptionNode(javadoc);
+        if (containsSummaryTagWithHtmlTag(javadoc)) {
+            summary = getInlineHtmlSentence(javadoc);
+        }
+        else if (descriptionNode == null) {
```

Comment:
this condition should be first in IF-ELSE block.
How it couldbe empty is we did check by `containsSummaryTag(ast)` at the beginning.

Reply:
Sir, it is not assured not containing Summary tag will have description. Eg : 
```
 /**
   * {@summary}
   * 
   */
  public void invalidInlineJavadocTwo() // violation
  {
  }
```